### ISSUE TITLE
[FEAT]: Please support fetching the secret asynchronously

### ISSUE THREAD

**Issue by justinmchase** [octokit/webhooks.net#486]
### Describe the need

I need to fetch the secret from Key Vault and it can change over the lifecycle of the container.

The code you have to verify the signature is to fetch the secret at configure time and not asynchronous...

```cs
public static IEndpointConventionBuilder MapGitHubWebhooks(
  this IEndpointRouteBuilder endpoints,
  string path = "/api/github/webhooks",
  string secret = null!
)
```

At startup we are initializing it like this...

```cs
public static WebApplication UseGitHubWebhooks(this WebApplication app)
{
  var githubOptions = app.Services.GetRequiredService<IOptions<GitHubOptions>>().Value;
  var secretClient = app.Services.GetRequiredService<ISecretClient>();
  var secret = secretClient.GetSecret(githubOptions.WebhookSecretKey).ConfigureAwait(false).GetAwaiter().GetResult();
  app.MapGitHubWebhooks($"{ApplicationName.BaseUrl}/v1{githubOptions.WebhookPath}", secret.Value);
  return app;
}
```

This works but there are a couple of problems with this:

1. Having to add an async call at this stage of the application startup isn't great.
2. Theoretically the secret can change in key vault but I'll never be able to update it here without restarting my pods, there is no dynamic fetching of the secret going on.

Now there is a couple of ways that you could solve this but in my opinion simply adding some kind of `IWebhookSecret` interface which has an async method `public Task<string> GetSecret()` would work great.

That way I could inject an instance of an object that knows how to reach out to my `ISecretClient` and dynamically fetch before validating.

```cs
var secret = await secretClient.GetSecret();
if (!await VerifySignatureAsync(context, secret, body).ConfigureAwait(false))
{
  Log.SignatureValidationFailed(logger);
  return;
}
```

### SDK Version

Octokit.net 11.0.1

### API Version

N/A

### Relevant log output

```shell
Not relevant to the API
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by JamieMagee** [comment]
Thanks for the issue. This sounds like a similar issue to #261 and #262. Could you take a look and let me know what you think?

**Comment by justinmchase** [comment]
Yes I think #262 would solve my ask exactly. Now I was imagining it was a service rather than just a function but an async function will work too.
